lib/std/time/TimeZone.zig

```diff
@@ -0,0 +1,315 @@
+    i = 0;
+    while (i < header.counts.isutcnt) : (i += 1) {
+        const ut = try reader.readByte();
+        if (ut == 1) {
+            timetypes[i].flags |= 0x04;
+            if (!timetypes[i].standardTimeIndicator()) return error.Malformed; // rfc8536: standard/wall value MUST be one (1) if the UT/local value is one (1)
+        }
+    }
+
+    // Footer
+    var footer: ?[]u8 = null;
+    if (!legacy) {
+        if ((try reader.readByte()) != '\n') return error.Malformed; // An rfc8536 footer must start with a newline
+        var footerdata_buf: [128]u8 = undefined;
+        const footer_mem = reader.readUntilDelimiter(&footerdata_buf, '\n') catch |err| switch (err) {
+            error.StreamTooLong => return error.OverlargeFooter, // Read more than 128 bytes, much larger than any reasonable POSIX TZ string
+            else => |e| return e,
+        };
+        if (footer_mem.len != 0) {
+            footer = try allocator.dupe(u8, footer_mem);
+        }
+    }
+    errdefer if (footer) |ft| allocator.free(ft);
+
+    return TimeZone{
+        .transitions = transitions,
+        .timetypes = timetypes,
+        .leapseconds = leapseconds,
+        .footer = footer,
+    };
+}
+
+pub fn deinit(self: *TimeZone, allocator: std.mem.Allocator) void {
+    if (self.footer) |footer| {
+        allocator.free(footer);
+    }
+    allocator.free(self.leapseconds);
+    allocator.free(self.transitions);
+    allocator.free(self.timetypes);
+}
+
+pub const Projection = struct {
+    /// Offset from UNIX timestamp including leap seconds.
+    offset: i32,
+    name_data: [6:0]u8,
+
+    pub fn name(self: *const TimeType) [:0]const u8 {
+        return std.mem.sliceTo(&self.name_data, 0);
+    }
+};
+
+/// Project UTC timestamp to this time zone.
+pub fn project(self: TimeZone, seconds: i64) Projection {
+    const index = index: {
+        var left: usize = 0;
+        var right: usize = self.transitions.len;
+
+        // Adapted from std.sort.binarySearch.
+        var mid: usize = 0;
+        while (left < right) {
+            // Avoid overflowing in the midpoint calculation
+            mid = left + (right - left) / 2;
+            // Compare the key with the midpoint element
+            switch (std.math.order(seconds, self.transitions[mid].ts)) {
+                .eq => break :index mid,
+                .gt => left = mid + 1,
+                .lt => right = mid,
+            }
+        }
+        break :index mid - @intFromBool(self.transitions[mid].ts > seconds);
+    };
+    const leap_adjustment = leap: {
+        var i = self.leapseconds.len;
+        if (i == 0) break :leap 0;
+        while (i > 0) {
+            i -= 1;
+            if (self.leapseconds[i].occurrence < seconds) break;
+        }
+        break :leap self.leapseconds[i].correction;
+    };
+    const tt = self.transitions[index].timetype;
+    return .{
+        .offset = tt.offset + leap_adjustment,
+        .name_data = tt.name_data,
```

Comment:
Not sure I get that one :) Are you proposing a leap second correction for a UTC offset? IIUC, time zones as defined in TZif specify offsets *relative* to a certain reference time.